### PR TITLE
NOJIRA make build.sbt loadable in intellij

### PR DESCRIPTION
**Pull request by oscarduignan**
Everyone has migrated to play 3 now

Took this chance to:
- bump the node version to latest LTS
- pin the exact nunjucks version
- explain why we need NODE_ENV=production when building
- switch the command used to install node_modules to `npm ci` for safety
- move better-npm-audit out of dependencies into devDependencies so we
  don't package it into the artefact and install it into services
- removed shared build config that was no longer needed without
  play 28 and play 29

### CODE REVIEW

File: build.sbt
Comment (oscarduignan) [inline build.sbt:83]: this change was because in testing, I noticed that running
```
nvm install && PLAY_VERSION=2.8 sbt clean +test
```
was not running the test twice - once for 2.12 and once for 2.13 - it was just running once

so it's a bit more wordy but with this approach we can run the tests in all cross compiled scala versions for the current play version with
```
nvm install && PLAY_VERSION=2.8 sbt clean "project itServer" +test
```

going to add that to readme but I'll share with other devs first for feedback

Comment (oscarduignan) [reply build.sbt:83]: this will need an update to build-jobs before merging too

File: project/LibDependencies.scala
Comment (oscarduignan) [inline project/LibDependencies.scala:32]: made this match what we have in play-frontend-hmrc - not sure what the M6 represented here, but assuming it was a early version with compatibility before the new package was created for 2.9 onwards as play-filters-helpers (with 3.0 also moving it to the new org too)

File: project/LibDependencies.scala
Comment (oscarduignan) [inline project/LibDependencies.scala:56]: this might have been causing some issues too I think, because we were importing play - not totally sure what the minimum required changes were to get intellij to load the build.sbt, we play-filters-helpers above instead

https://github.com/playframework/playframework/blob/a4615d8c2d020dfd38ee5e1c9a467e5357d8c4b2/dev-mode/sbt-plugin/src/main/scala/play/sbt/PlayImport.scala#L59

File: project/project/plugins.sbt
Comment (oscarduignan) [inline project/project/plugins.sbt:1]: maybe created by mistake or left behind following a change, we no longer use sbt-play-cross-compilation

File: project/plugins.sbt
Comment (oscarduignan) [inline project/plugins.sbt:14]: moved at some point https://github.com/sbt/sbt-uglify

File: project/plugins.sbt
Comment (oscarduignan) [inline project/plugins.sbt:6]: no point cross building a library that changes so little, if a new play version comes along, just update it and if you need to patch old versions while people are still updating then you can do that

File: play-nunjucks-play-30/src/main/scala/uk/gov/hmrc/nunjucks/NunjucksModule.scala
Comment (oscarduignan) [inline play-nunjucks-play-30/src/main/scala/uk/gov/hmrc/nunjucks/NunjucksModule.scala:28]: some deprecation warnings

File: play-nunjucks-play-30/package.json
Comment (oscarduignan) [inline play-nunjucks-play-30/package.json:10]: with it in dependencies, and the way were installing, better-npm-audit would possibly be getting added to the artefact

File: build.sbt
Comment (oscarduignan) [inline build.sbt:44]: merged all these since now only building for one play version

File: build.sbt
Comment (oscarduignan) [inline build.sbt:33]: moved this into here, since we don't want to publish integration test sub project

File: build.sbt
Comment (oscarduignan) [inline build.sbt:17]: think it's the above two are the default now, but just to be explicit, these are just used during build when we download node modules - won't impact services, they will still be using the trireme? dependency to run nunjucks js in jvm

File: build.sbt
Comment (oscarduignan) [inline build.sbt:17]: not copied this to playNunjucksPlay30 because we do want to publish the "play-nunjucks-play-30" project

Comment (JoPintoPaul) [reply build.sbt:17]: @oscarduignan Does this mean the artefact name will change from `play-nunjucks-play-30` back to `play-nunjucks`?

File: project/plugins.sbt
Comment (JoPintoPaul) [reply project/plugins.sbt:14]: @oscarduignan There's a bobby warning about this version of `sbt-sasify`:
```
[0m[[0m[33mwarn[0m] [0m[0mWARNING: Your build has 1 bobby warning(s). Please take action to fix these before the listed date, or they will become violations that fail your build[0m
[0m[[0m[33mwarn[0m] [0m[0m (1) org.irundaia.sbt:sbt-sassify (1.4.11)[0m
[0m[[0m[33mwarn[0m] [0m[0m     Reason: To provide support for Apple Silicon move to io.github.irundaia 1.5.2 or higher
```

**Comment by JoPintoPaul** [comment]
@oscarduignan Given all the changes to remove multi-version builds, should we also consider renaming the existing folders and build file references? At the moment it seems like we're doing a bit of both, without other Play versions being used in the project the `-play-30` suffix feels like a smell.

File: build.sbt
Comment (oscarduignan) [reply build.sbt:17]: it stays as play-nunjucks-play-30, because I've left the play nunjucks project as
```
Project("play-nunjucks-play-30", file("play-nunjucks-play-30"))
```
so it has the [id](https://github.com/sbt/sbt/blob/1.10.x/main/src/main/scala/sbt/Project.scala#L172) explicitly defined, rather than using [the project macro which uses the name of the val as the id](https://github.com/sbt/sbt/blob/1.10.x/main/src/main/scala/sbt/Project.scala#L998)

possibly still be annoying for future maintainers to have the folder have the version number in it, probably be better to have just play-nunjucks so never have to rename it, or hoist the src dir up again to top level, but those are big enough changes that I would do them later and probably leave it up to future maintainers

File: project/plugins.sbt
Comment (oscarduignan) [reply project/plugins.sbt:14]: ah I'll bump this thank you!

**Comment by oscarduignan** [comment]
I'll rename the dirs to remove the -play-30 suffix, that will definitely make the future changes easier as new versions are created where folder renames are just a pain in the butt for looking at git history, but I think we should leave the artefact names suffixes so that pr-commenter never suggests someone to move to version they shouldn't

File: README.md
Comment (JoPintoPaul) [inline README.md:292]: @oscarduignan Given that we're dropping Play 2.8 and Play 2.9 support, should be update this text?

File: build.sbt
Comment (oscarduignan) [reply build.sbt:17]: have renamed the folders now but kept it a separate commit to make it easier to review just the changes

File: project/LibDependencies.scala
Comment (oscarduignan) [reply project/LibDependencies.scala]: removed support for play28 and play29 now because no-one using them anymore, everyone on play30 so this is no longer relevant

File: project/plugins.sbt
Comment (oscarduignan) [reply project/plugins.sbt:14]: done

File: build.sbt
Comment (oscarduignan) [inline build.sbt:16]: bit of context on this, we still have two sub-projects, but we can just run "sbt test" to run the integration server tests, we can do this because without any aggregate defined [sbt will create a default root project that aggregates all the sub projects](https://www.scala-sbt.org/1.x/docs/Multi-Project.html#Default+root+project)

Comment (oscarduignan) [reply build.sbt:16]: so no need for us to keep the aggregate library project after dropping support for other play versions, and with the idea that in the future people probably won't want to maintain support for 2 at once, when you can just patch the old version if you need to

**Comment by oscarduignan** [comment]
I've run the it-server locally and clicked around and didn't run into any issues

File: README.md
Comment (oscarduignan) [reply README.md]: done!

File: docs/getting-started-mdtp.md
Comment (ellamdav) [inline docs/getting-started-mdtp.md:17]: can remove this Play 2.6 reference?

Comment (oscarduignan) [reply docs/getting-started-mdtp.md]: done 🙌

**Review by ellamdav — APPROVED** [review]

**Pull request merged**
